### ISSUE TITLE
Craft 4 - Integrity violation after saving entry with link field (in matrix)

### ISSUE THREAD

**Issue by Antimated** [sebastian-lenz/craft-linkfield#219]
Trying to save a link field (on an existing craft install that's been updated to craftcms 4.0.0.1) results in a integrity violation error. This error occurs mostly when switching from/to different link types.

![Screenshot 2022-05-05 at 15 05 01](https://user-images.githubusercontent.com/2869409/166928992-1516e078-c688-48f9-b460-02ea9b67bfcb.png)

**Comment by sebastian-lenz** [comment]
The error does not contain a reference to the link field, also the affected table is not the link field table. Why do you assume that error is related to this plugin?

**Comment by Antimated** [comment]
> The error does not contain a reference to a link field, also the affected table is not the link field table. Why do you assume that error is related to this plugin?

Because whenever I update this field the error occurs :/

**Comment by sebastian-lenz** [comment]
I've received another issue from a user who also complains about a similar foreign key issue, I guess the new CMS version exposes issues within the data structure of certain projects. In you example, the database complains about an element with the id `166` not being present in the table `craft_elements`. Can you check whether the element record is there? Also, does the element select dialog work in other instances, e.g. for reference fields?

**Comment by Antimated** [comment]
> I've received another issue from a user who also complains about a similar foreign key issue, I guess the new CMS version exposes issues within the data structure of certain projects. In you example, the database complains about an element with the id `166` not being present in the table `craft_elements`. Can you check whether the element record is there?

No, doesn't seem like that element is present for some reason...
<img width="1084" alt="Screenshot 2022-05-05 at 15 42 09" src="https://user-images.githubusercontent.com/2869409/166936191-4f66c8eb-dcdb-427e-bae4-b6085ab67618.png">

>  Also, does the element select dialog work in other instances, e.g. for reference fields?
Yes I added a simple "entries" field with these sources (it's a relatively empty site) and the element select dialog seems to work there.

![Screenshot 2022-05-05 at 15 44 42](https://user-images.githubusercontent.com/2869409/166936614-68107671-bdfd-45dc-a6ac-c0a53942ef6a.png)

**Comment by sebastian-lenz** [comment]
Okay, I think I've found the bug, it's related to the way Craft now handles newly created elements. I've published a new release that contains the fix.
